drv/stm32h7-update-api/src/lib.rs

```diff
@@ -26,4 +28,35 @@ pub const BLOCK_SIZE_BYTES: usize = FLASH_WORD_BYTES * FLASH_WORDS_PER_BLOCK;
 
 pub const BLOCK_SIZE_WORDS: usize = BLOCK_SIZE_BYTES / 4;
 
+#[derive(
+    Clone,
+    Copy,
+    Eq,
+    PartialEq,
+    FromPrimitive,
+    Serialize,
+    Deserialize,
+    SerializedSize,
+)]
+pub enum SlotId {
+    Active = 0,
+    Inactive = 1,
+}
+
+impl TryFrom<u16> for SlotId {
```

Comment:
If you're using this for something other than IPC, carry on.

For IPC specifically, your use of `encoding: Hubpack` relieves you from needing these instances (and explicit `= X` discriminator values). FYI.